android/app/src/main/java/app/organicmaps/routing/RoutingController.java

```diff
@@ -266,15 +257,18 @@ public void attach(@NonNull Container container)
   }
 
   @Override
-  public void initialize(@Nullable Void aVoid)
+  public void initialize(@NonNull Context context)
   {
     mLastRouterType = Framework.nativeGetLastUsedRouter();
     mInvalidRoutePointsTransactionId = Framework.nativeInvalidRoutePointsTransactionId();
     mRemovingIntermediatePointsTransactionId = mInvalidRoutePointsTransactionId;
 
     Framework.nativeSetRoutingListener(mRoutingListener);
     Framework.nativeSetRouteProgressListener(mRoutingProgressListener);
-    Framework.nativeSetRoutingRecommendationListener(mRoutingRecommendationListener);
+    Framework.nativeSetRoutingRecommendationListener(recommendation -> UiThread.run(() -> {
+      if (recommendation == Framework.ROUTE_REBUILD_AFTER_POINTS_LOADING)
+        setStartPoint(LocationHelper.from(context).getMyPosition());
```

Comment:
Does anyone know the purpose of RoutingRecommendationListener callback? This is the only controversal part in this patch...

Reply:
```
  if (m_routeRecommendCallback != nullptr)
  {
    // Check if we've found my position almost immediately after route points loading.
    auto constexpr kFoundLocationInterval = 2.0;
    auto const elapsed = chrono::steady_clock::now() - m_loadRoutePointsTimestamp;
    auto const sec = chrono::duration_cast<chrono::duration<double>>(elapsed).count();
    if (sec <= kFoundLocationInterval)
    {
      m_routeRecommendCallback(Recommendation::RebuildAfterPointsLoading);
      CancelRecommendation(Recommendation::RebuildAfterPointsLoading);
    }
  }
```

Reply:
What I see in the code:
MwmActivity.onSafeCreate->RestoreRouteTask.restoreRoute->LoadRoutePoints->m_loadRoutePointsTimestamp->SetUserCurrentPosition->m_routeRecommendCallback

I suppose the UX logic is: in navigation mode (some route exists), the app can close activity (go to background). When _foregrounding_ the app again, it loads a saved (previous) route and calls callback immediately after getting a location after loading the route.

Reply:
ok, the current approach should work, but I want to remove static members and global RoutingController on the next step.